src/BloomExe/web/ServerBase.cs

```diff
@@ -116,117 +74,122 @@ private bool StartWithExceptionHandling(out Exception error)
 			catch (Exception e)
 			{
 				error = e;
-				_listener = null;
 				return false;
 			}
 		}
 
 		private bool TryStart()
 		{
-			_listener = new HttpListener();
 			_listener.AuthenticationSchemes = AuthenticationSchemes.Anonymous;
 			_listener.Prefixes.Add(PathEndingInSlash);
 			_listener.Start();
-			//			_listener.BeginGetContext(new AsyncCallback(ListenerCallback), _listener);
 
-			_listenerThread = new Thread(HandleRequests);
 			_listenerThread.Start();
 
+			for (var i = 0; i < _workers.Length; i++)
+			{
+				_workers[i] = new Thread(WaitForRequests);
+				_workers[i].Start();
+			}
+
 			return GetIsAbleToUsePort();
 		}
 
-		private bool GetIsAbleToUsePort()
+		private void HandleRequests()
 		{
-			IsAbleToUsePort = false;
-			try
+			while (_listener.IsListening)
 			{
-				var x = new WebClientWithTimeout { Timeout = 3000 };
+				var context = _listener.BeginGetContext(ContextReady, null);
 
-				IsAbleToUsePort = ("OK" == x.DownloadString(PathEndingInSlash + "testconnection"));
-			}
-			catch (Exception e)
-			{
-				IsAbleToUsePort = false;
+				if (0 == WaitHandle.WaitAny(new[] { _stop, context.AsyncWaitHandle }))
+					return;
 			}
-			return IsAbleToUsePort;
 		}
 
-		private void HandleRequests()
+		private void ContextReady(IAsyncResult ar)
```

Comment:
I don't know where this name came from...maybe it historically did something a bit different? Now it seems a better name would be something like QueueRequest, maybe with a comment that this method is called (in the _listenerThread?) when we obtain an HTTP request from the _listener, and queues it for processing by a worker.